mars/dataframe/datasource/from_tensor.py

```diff
@@ -49,26 +49,32 @@ def _set_inputs(self, inputs):
         self._input = inputs[0]
 
     def __call__(self, input_tensor, index, columns):
-        if index is not None or columns is not None:
-            if input_tensor.shape != (len(index), len(columns)):
-                raise ValueError(
-                    '({0},{1}) should have the same shape with tensor: {2}'.format(index, columns, input_tensor.shape))
-        if input_tensor.ndim == 1:
-            # convert to 1-d DataFrame
-            index_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[0]))
-            columns_value = parse_index(pd.RangeIndex(start=0, stop=1))
-        elif input_tensor.ndim != 2:
+        if input_tensor.ndim != 1 and input_tensor.ndim != 2:
             raise ValueError('Must pass 1-d or 2-d input')
-        else:
-            # convert to DataFrame
-            index_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[0]))
-            columns_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[1]))
 
-        # overwrite index_value and columns_value if user has set them
         if index is not None:
+            if input_tensor.shape[0] != len(index):
+                raise ValueError(
+                    'index {0} should have the same shape with tensor: {1}'.format(index, input_tensor.shape[0]))
+            if not isinstance(index, pd.Index):
```

Comment:
I think for now we can convert `index` to pandas Index when it is `pd.Index` and not `mars.core.Base`, we could add support for mars objects later.

Reply:
> we can convert `index` to pandas Index when it is `pd.Index` and not `mars.core.Base`

Did you mean "converting mars IndexValue to pandas Index"?

Reply:
No, I meant that Mars tileable objects like tensor are not considered here.

Reply:
We don't consider mars tileable here. `not instance(index, pd.Index)` means the `index` can be a `list`, or `np.ndarray`.

Reply:
Yeah, but tileable objects are not `pd.Index` as well. We could raise `NotImplementedError` here, or they would be converted to `pd.Index`.

Reply:
I see, fixed now.